<launch>
  <arg name="first_tb3"  default="carter1"/>
  <arg name="second_tb3" default="carter2"/>
  <arg name="third_tb3" default="carter3"/>

  <group ns="carter1/map_merge">
    <param name="init_pose_x"   value="1.36"/>
    <param name="init_pose_y"   value="-7.8"/>
    <param name="init_pose_z"   value="0"/>
    <param name="init_pose_yaw" value="-1.57"  />
  </group>

    <group ns="carter2/map_merge">
    <param name="init_pose_x"   value="0"/>
    <param name="init_pose_y"   value="0"/>
    <param name="init_pose_z"   value="0"/>
    <param name="init_pose_yaw" value="1.57"  />
  </group>
    <group ns="carter3/map_merge">
    <param name="init_pose_x"   value="4"/>
    <param name="init_pose_y"   value="0"/>
    <param name="init_pose_z"   value="0"/>
    <param name="init_pose_yaw" value="1.57"  />
  </group>

  <!-- args="0 0 0 1.57 0 0  
  args="1.34 0 0 1.57 0 0
  args="5.54 0 0 1.57 0 0 -->
  <!-- <group ns="carter2/map_merge">
    <param name="init_pose_x"   value="-30.06"/>
    <param name="init_pose_y"   value="-8"/>
    <param name="init_pose_z"   value="0"/>
    <param name="init_pose_yaw" value="0"  />
  </group>

  <group ns="carter3/map_merge">
    <param name="init_pose_x"   value="15.94"/>
    <param name="init_pose_y"   value="-34"/>
    <param name="init_pose_z"   value="0"/>
    <param name="init_pose_yaw" value="0"  />
  </group> -->

  <node pkg="multirobot_map_merge" type="map_merge" respawn="false" name="map_merge" output="screen">
    <param name="robot_map_topic" value="map"/>
    <param name="robot_map_updates_topic" value="map_updates"/>
    <param name="robot_namespace" value="carter"/>
    <param name="merged_map_topic" value="map_merge_topic"/>
    <param name="world_frame" value="map_merge_frame"/>
    <param name="known_init_poses" value="true"/>
    <param name="merging_rate" value="1"/>
    <param name="discovery_rate" value="0.5"/>
    <param name="estimation_rate" value="0.1"/>
    <param name="estimation_confidence" value="1.5"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg first_tb3)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg first_tb3)/map 100"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg second_tb3)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg second_tb3)/map 100"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg third_tb3)_tf_broadcaster" args="0 0 0 0 0 0 /map /$(arg third_tb3)/map 100"/>
  <node pkg="tf" type="static_transform_publisher" name="map_merge_tf_broadcaster" args="5.15 -0.38 0 0 0 0 map map_merge_frame 100"/>
</launch>